### PR TITLE
Add pre-commit and bunch of autofixes support

### PR DESCRIPTION
**Pull request by Avasam**
Here's the "magically autofixing on pull-request" I was talking about.
You install `pre-commit` locally with `pip install pre-commit` and run it locally using `pre-commit run [--all-files]` (without `--all-files` it only runs on changed files, which is what would happen on the CI).
You can still run all commands individually if you feel like it (like `black .`)

Added autoformatting for toml, yaml and ini files as well (not json cuz there's none in this project).
Autofixes for mixed-line-ending, case-conflict trailing whitespaces and end-of-file newline. Although I didn't activate those last two because it'd create a massive commit, better keep that for it's own, easily ignorable from history.

Users who enjoy pre-commit git hooks (I don't, some do), can also now use this as a pre-commit hook on their own accord.

If this is accepted, the next step would be to sign-in on https://pre-commit.ci/ and connect it to this repo. (it's the CI autofixes version of https://pre-commit.com)

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -1,7 +1,6 @@
 # A clang-format style that approximates Python's PEP 7
 # Useful for IDE integration
 BasedOnStyle: Google
-AlwaysBreakAfterReturnType: All
```
Comment (Avasam) [inline .clang-format:4]: It found a duplicate entry!

File: com/win32comext/mapi/src/mapi_headers/MAPIOID.h [hunk 2]
```diff
@@ -1,106 +1,107 @@
-/*
- *  M A P I O I D . H
- *
- *  MAPI OID definition header file
- *
- *  Copyright (c) 2009 Microsoft Corporation. All Rights Reserved.
```
Comment (Avasam) [inline com/win32comext/mapi/src/mapi_headers/MAPIOID.h:6]: It found a weird character and fixed line endings

File: pycln.toml [hunk 3]
```diff
@@ -20,5 +20,5 @@ skip_imports = [
   "win32pipe",
   "win32ras",
   "win32ui",
-  "winxpgui",
+  "winxpgui"
```
Comment (Avasam) [inline pycln.toml:23]: That one I'd rather not, but it isn't currently configurable: https://github.com/macisamuele/language-formatters-pre-commit-hooks/issues/152

File: com/win32comext/mapi/src/mapi_headers/MAPIOID.h (hunk 2 above)
Comment (mhammond) [reply com/win32comext/mapi/src/mapi_headers/MAPIOID.h:6]: We should exclude these "vendored" files though?

**Review by mhammond — CHANGES_REQUESTED** [review]
Thanks, but files copied in from external sources should be excluded from these rules.

**Review by mhammond — APPROVED** [review]
Thanks!

**Pull request merged**